library/func_misc.py

```diff
@@ -75,3 +74,76 @@ def Call(self, pos_args, named_args):
 
         # Equivalent to no return value?
         return value.Null
+
+
+class StartsWith(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        spec = ArgsSpec([value_e.Str, value_e.Str], {})
```

Comment:
Actually one thing I noticed is that you have to do `value_e.Str` here, and then `cast(value.Str` later

I wonder if we can come up with an API without that

It might be more imperative, like

```
mystr = spec.Str(pos_args[0])
myint = spec.Int(pos_args[1])
spec.NoMorePositional()
str2 = spec.StrN(named_args, 'name2')
int3 = spec.IntN(named_args, 'name3')
spec.NoMoreNamed()
```

Of course this has to translate to C++, which makes it a bit trickier

We could also try a code gen solution

Something to think about

Reply:
Yes, I also found that unsatisfying.

Handling signatures like `max()` (I know we are implementing it in YSH, but still) makes it even trickier. This is definitely something I plan to iterate on as we implement more builtin functions.